[5.4] Automated Core Update Client

SniperSister opened the pull request:
### Summary of Changes
This PR implements automated core updates for Joomla. It's the "client" implementation, the server implementation can be found here: https://github.com/joomla-projects/Automated-Updates-Server

In general, the implemented concept utilizes existing logic and functionality ans has been built as a thin "remote control" layer around the current code:
* Update retrieval is handled by TUF, guaranteeing reliable, trustworthy update information. The information about a new version is therefore not "pushed" by the autoupdate server (causing potential loopholes if the server is compromised) but fetched by the site itself
* Update download and checksum handling is performed by the existing model, that com_joomlaupdate uses
* Update extraction is performed using the standalone extraction script of com_joomlaupdate - as this script normally resides in the administrator folder and that folder is often protected by additional measures (IP blocks, basic auth checks), a statement in the main index.php has been ended to require that script directly, allowing direct access
* Postinstall scripts are again performed by the existing com_joomlaupdate models

The communication between the update server (which handles periodic health checks and triggers the updates) and the site happens via a bunch of newly added webservice endpoints. For access control, an auth token, that is generated in the site and is sent to the server on registration, is used.

Besides these endpoints, the PR adds multiple "supporting" extensions and tweaks:
* a postinstall message, highlighting the new feature on existing installations, pushing site owners to enable it
* a guided tour, highlighting the new feature too
* a quick icon, displaying the current feature and connection status
* new mail templates to notify admins about successful or failed updates
* CLI commands to register and unregister a site in the autoupdate server.

This PR is joint effort together with @rdeutz @bembelimen @HLeithner - thank you guys! Thank you @brianteeman for the language support :) and thank you @richard67 for taking care of  the adjustments to 5.4

### Testing Instructions
The following instructions will cover a full autoupdate cycle. In order for the update to be performed, you have to install the site on a **publicly accessible server**. **A local enviroment will not work**.

1a. Install a new Joomla site using the package provided at [Full Download](https://update.joomla.org/tuftest/Joomla_5.4.105-Stable-Full_Package.zip)
1b. Update a Joomla site using the package provided at [Update Download](https://update.joomla.org/tuftest/Joomla_5.4.105-Stable-Update_Package.zip)
1c. Update without automated update, change in the database table `#_update_sites` the column `location` to `https://update.joomla.org/alpha/`, change in the file `administrator/components/com_joomlaupdate/src/Model/UpdateModel.php` in line 123 to `$updateURL = 'https://update.joomla.org/alpha/';`
4. Log into the administrator site and check that the guided tour for the new feature is shown; that login will also trigger the registration of your site in the update server
5. As a (faked) newer version is available, the update server will trigger an autoupdate of your newly installed site automatically. This process will take approx. 5-10 minutes.
6. When the update is completed (or failed) you'll receive an email to the address provided during the installation process
7. Verify the update result after the mail has been received
8. Manually disable autoupdates using the provided option in com_joomlaupdate; verify the registration state using the quickicon in the dashboard
9. Re-Enable autoupdates and verify the state again


### Link to documentations
Please select:
- [x] Documentation link for docs.joomla.org: Todo
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

bembelimen committed:
Add extension class to joomlaupdater

bembelimen committed:
Allow components to change config form

bembelimen committed:
Add extension class for joomlaupdater

bembelimen committed:
Add needed config fields for automated updates

bembelimen committed:
Add initial joomlaupdate webservice plugin

bembelimen committed:
Implement healthcheck API

bembelimen committed:
Move token check to header

bembelimen committed:
Update last health check date in extension table

bembelimen committed:
Implement token generation on config save

bembelimen committed:
Add deploy version

bembelimen committed:
Add joomlaupdate webservice plugin as core extension

bembelimen committed:
Add missing comment

bembelimen committed:
Fix comment

bembelimen committed:
Add type hint for model

bembelimen committed:
Return error in health check when auto update is disabled

bembelimen committed:
Move primary method to top in health check

bembelimen committed:
Implement getUpdate webservice call

bembelimen committed:
Make view call more generic

bembelimen committed:
Remove unused code

bembelimen committed:
Rename parameter

bembelimen committed:
Update joomlaupdate config

bembelimen committed:
Reimplement token generation

bembelimen committed:
Take care of minimum stability

bembelimen committed:
Add restricter rule when auto updater is disabled

bembelimen committed:
Fix yntax error

bembelimen committed:
Update getUpdate to use core functionality

bembelimen committed:
Improve update disable information

bembelimen committed:
Fix notice

bembelimen committed:
Fix notice

bembelimen committed:
Correct ID

bembelimen committed:
Add prepareUpdate

bembelimen committed:
Add update finalization

bembelimen committed:
Convert prepareUpdate and finalizeUpdate to POST

bembelimen committed:
Fix code style

bembelimen committed:
Code style fix

Commit:
Implement update server registration.

Commit:
Merge remote-tracking branch 'au/automated-updates/client' into automated-updates/register

# Conflicts:
#	administrator/components/com_joomlaupdate/src/Extension/JoomlaupdateComponent.php

HLeithner committed:
Update components/com_config/src/Service/Router.php

HLeithner committed:
Update libraries/src/Version.php

HLeithner committed:
Update libraries/src/Updater/ConstraintChecker.php

HLeithner committed:
Update installation/src/Helper/DatabaseHelper.php

HLeithner committed:
Update components/com_config/src/Service/Router.php

bembelimen committed:
Merge branch '5.3-dev' into automated-updates/client

bembelimen committed:
Merge branch 'automated-updates/client' into automated-updates/register

bembelimen committed:
Implement quickicon

SniperSister committed:
Merge pull request #3 from joomla-projects/automated-updates/dashboard-icon

Implement Quickicon

SniperSister committed:
Merge pull request #2 from joomla-projects/automated-updates/register

Implement update server registration.

SniperSister committed:
use correct site url

SniperSister committed:
fix input

Commit:
Merge remote-tracking branch 'upstream/5.3-dev' into automated-updates/sync

HLeithner committed:
Merge pull request #7 from joomla-projects/automated-updates/sync

Automated updates/sync

Commit:
Rework register

SniperSister committed:
Merge pull request #8 from joomla-projects/automated-updates/register-rework

Rework register

SniperSister committed:
Add entry-point if-case for autoupdate extraction process

SniperSister committed:
Merge remote-tracking branch 'upstream/5.3-dev' into automated-updates/client

brianteeman committed:
Update JoomlaupdateComponent.php

SniperSister committed:
Merge pull request #10 from brianteeman/patch-1

Update JoomlaupdateComponent.php

SniperSister committed:
Added joomlaupdate webservice plugin to install and update SQL

SniperSister committed:
Merge branch 'automated-updates/client' of github.com:joomla-projects/Automated-Updates-Client into automated-updates/client

bembelimen committed:
Enhance guided tours for Joomla 5.3 with new automated updates feature and updated descriptions

SniperSister committed:
Refactor registration logic; move to model

SniperSister committed:
Merge pull request #11 from joomla-projects/automated-updates/client-guided-tour

Enhance guided tours for Joomla 5.3 with new automated updates feature and updated descriptions

SniperSister committed:
fix result

SniperSister committed:
Added acl check for autoupdate registration

SniperSister committed:
fix cast

SniperSister committed:
Fix acl check

bembelimen committed:
Merge pull request #12 from joomla-projects/feature/refactored-registration

Refactored Registration

Commit:
Add Postinstallation Message

SniperSister committed:
Merge pull request #13 from joomla-projects/automated-updates/postmessage

Add Postinstallation Message

Commit:
Update Postinstallation action

HLeithner committed:
Merge pull request #14 from joomla-projects/automated-updates/postmessage

Update Postinstallation action

brianteeman committed:
fix copyright an subpackage

simple copy paste error

SniperSister committed:
Merge pull request #15 from brianteeman/patch-1

fix copyright an subpackage

brianteeman committed:
Update com_joomlaupdate.ini

Throughout Joomla we avoid the word "click" as you can only click with a mouse and not with a screen reader or phone. Usually we use the word "select"

SniperSister committed:
Merge pull request #16 from brianteeman/patch-2

Update com_joomlaupdate.ini

SniperSister committed:
remove unused variable

SniperSister committed:
fix default value for minimum_stability

SniperSister committed:
cs fix

bembelimen committed:
Prevent update notification emails when auto updater is active

SniperSister committed:
Merge pull request #20 from joomla-projects/automated-updates/disable-notification-email

Prevent update notification emails when auto updater is active

SniperSister committed:
Call cleanup method and trigger actionlog

HLeithner committed:
Merge pull request #21 from joomla-projects/feature/trigger-aciton-log-and-cleanup

Call cleanup method and trigger actionlog

SniperSister committed:
Adjust actionlog item for automated updates

SniperSister committed:
added tempaltes to initial dataset

SniperSister committed:
added update sqls

SniperSister committed:
Update administrator/language/en-GB/plg_actionlog_joomla.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

SniperSister committed:
added mailtemplate texts

SniperSister committed:
remove irrelevant change

SniperSister committed:
cs fixes

SniperSister committed:
re-add defines path

SniperSister committed:
Merge pull request #22 from joomla-projects/feature/actionlog-item-fixes

Adjust actionlog item for automated updates

SniperSister committed:
added quickicon to extension helper

SniperSister committed:
remove unused check

SniperSister committed:
cs fix

SniperSister committed:
Merge branch 'automated-updates/client' of github.com:joomla-projects/Automated-Updates-Client into automated-updates/client

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,18 +3,30 @@
 ; License GNU General Public License version 2 or later; see LICENSE.txt
 ; Note : All ini files need to be saved as UTF-8
 
+COM_JOOMLAUPDATE_AUTOUPDATE_REGISTER_SUCCESS="Successful registered to autoupdate service"
```
We have removed all uses of the redundant word "successful"
```suggestion
COM_JOOMLAUPDATE_AUTOUPDATE_REGISTER_SUCCESS="Registered to autoupdate service"
```

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,18 +3,30 @@
 
+COM_JOOMLAUPDATE_AUTOUPDATE_REGISTER_SUCCESS="Successful registered to autoupdate service"
+COM_JOOMLAUPDATE_AUTOUPDATE_REGISTER_ERROR="Error while registering to autoupdate service: %s (%d)"
+COM_JOOMLAUPDATE_AUTOUPDATE_UNREGISTER_SUCCESS="Successful unregistered from autoupdate service"
```
```suggestion
COM_JOOMLAUPDATE_AUTOUPDATE_UNREGISTER_SUCCESS="Unregistered from autoupdate service"
```

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,18 +3,30 @@
 COM_JOOMLAUPDATE_CAPTIVE_HEADLINE="Confirm your credentials"
 COM_JOOMLAUPDATE_CHECKED_UPDATES="Checked for updates."
+COM_JOOMLAUPDATE_CONFIG_AUTO_UPDATE_DISABLED_LABEL="<span class=\"fa-fw me-2 fa fa-info-circle\"></span> Automated updates are only supported for the \"default\" channel and \"Minimum Stability\" \"Stable\"."
+COM_JOOMLAUPDATE_CONFIG_AUTOMATED_UPDATES_LABEL="Automated updates"
```
The UI language styleguide says that labels should be capitalised

```suggestion
COM_JOOMLAUPDATE_CONFIG_AUTOMATED_UPDATES_LABEL="Automated Updates"
```

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,18 +3,30 @@
 COM_JOOMLAUPDATE_CONFIG_SOURCES_DESC="Configure where Joomla gets its update information from."
 COM_JOOMLAUPDATE_CONFIG_SOURCES_LABEL="Update Source"
 COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_CUSTOM="Custom URL"
+COM_JOOMLAUPDATE_CONFIG_UPDATE_LAST_CHECK_LABEL="Last checked"
```
```suggestion
COM_JOOMLAUPDATE_CONFIG_UPDATE_LAST_CHECK_LABEL="Last Checked"
```

brianteeman commented on administrator/language/en-GB/plg_webservices_joomlaupdate.ini:
```diff
@@ -0,0 +1,7 @@
+; Joomla! Project
+; (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
; (C) 2025 Open Source Matters, Inc. <https://www.joomla.org>
```

brianteeman commented on administrator/language/en-GB/plg_webservices_joomlaupdate.sys.ini:
```diff
@@ -0,0 +1,7 @@
+; Joomla! Project
+; (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
; (C) 2025 Open Source Matters, Inc. <https://www.joomla.org>
```

brianteeman commented on plugins/quickicon/autoupdate/autoupdate.xml:
```diff
@@ -0,0 +1,34 @@
+<extension type="plugin" group="quickicon" method="upgrade">
+	<name>plg_quickicon_autoupdate</name>
+	<author>Joomla! Project</author>
+	<creationDate>2025-01</creationDate>
```
```suggestion
	<creationDate>2025-03</creationDate>
```

brianteeman commented on plugins/quickicon/autoupdate/autoupdate.xml:
```diff
@@ -0,0 +1,34 @@
+	<license>GNU General Public License version 2 or later; see LICENSE.txt</license>
+	<authorEmail>admin@joomla.org</authorEmail>
+	<authorUrl>www.joomla.org</authorUrl>
+	<version>5.0.0</version>
```
```suggestion
	<version>5.3.0</version>
```

brianteeman commented on plugins/webservices/joomlaupdate/joomlaupdate.xml:
```diff
@@ -0,0 +1,21 @@
+<extension type="plugin" group="webservices" method="upgrade">
+	<name>plg_webservices_joomlaupdate</name>
+	<author>Joomla! Project</author>
+	<creationDate>2024-11</creationDate>
```
```suggestion
	<creationDate>2025-03</creationDate>
```

brianteeman commented on plugins/webservices/joomlaupdate/joomlaupdate.xml:
```diff
@@ -0,0 +1,21 @@
+	<name>plg_webservices_joomlaupdate</name>
+	<author>Joomla! Project</author>
+	<creationDate>2024-11</creationDate>
+	<copyright>(C) 2024 Open Source Matters, Inc.</copyright>
```
```suggestion
	<copyright>(C) 2025 Open Source Matters, Inc.</copyright>
```

brianteeman commented on plugins/webservices/joomlaupdate/joomlaupdate.xml:
```diff
@@ -0,0 +1,21 @@
+	<license>GNU General Public License version 2 or later; see LICENSE.txt</license>
+	<authorEmail>admin@joomla.org</authorEmail>
+	<authorUrl>www.joomla.org</authorUrl>
+	<version>5.0.0</version>
```
```suggestion
	<version>5.3.0</version>
```

brianteeman commented on plugins/webservices/joomlaupdate/services/provider.php:
```diff
@@ -0,0 +1,46 @@
+ * @package     Joomla.Plugin
+ * @subpackage  Webservices.Joomlaupdate
+ *
+ * @copyright   (C) 2023 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2025 Open Source Matters, Inc. <https://www.joomla.org>
```

brianteeman commented on plugins/webservices/joomlaupdate/src/Extension/Joomlaupdate.php:
```diff
@@ -0,0 +1,67 @@
+ * @package     Joomla.Plugin
+ * @subpackage  Webservices.joomlaupdate
+ *
+ * @copyright   (C) 2020 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2025 Open Source Matters, Inc. <https://www.joomla.org>
```

SniperSister committed:
fix js codestyles

brianteeman commented on administrator/components/com_joomlaupdate/config.xml:
```diff
@@ -72,4 +72,51 @@
+			layout="joomla.form.field.radio.switcher"
+			showon="updatesource:default[AND]minimum_stability:4"
+			validate="options"
+		>
```
```suggestion
			>
```

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,18 +3,30 @@
 COM_JOOMLAUPDATE_CONFIG_SOURCES_LABEL="Update Source"
 COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_CUSTOM="Custom URL"
+COM_JOOMLAUPDATE_CONFIG_UPDATE_LAST_CHECK_LABEL="Last checked"
+COM_JOOMLAUPDATE_CONFIG_UPDATE_LAST_CHECK_DESC="The last time the automated update server could reach this site. If this date is too old, something is wrong with the automated update and you should re-initiate the process by saving the configuration."
```
its a bit clunky. what is "too old"?

```suggestion
COM_JOOMLAUPDATE_CONFIG_UPDATE_LAST_CHECK_DESC="The last time the automated update server could reach this site. If this date is too old, something is wrong with the automated update and you should re-initiate the process by re-saving the configuration."
```

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,18 +3,30 @@
 COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_DEFAULT="Default"
 COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_LABEL="Update Channel"
 COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_NEXT="Joomla Next"
+COM_JOOMLAUPDATE_CONFIG_UPDATE_TOKEN_DESC="This is the update token that Joomla uses to authenticate with the Joomla Update Server. It is used to ensure that only your site can update itself."
```
would it be better to say??
```suggestion
COM_JOOMLAUPDATE_CONFIG_UPDATE_TOKEN_DESC="This is the update token that this site uses to authenticate with the Joomla Update Server. It is used to ensure that only your site can update itself."
```

brianteeman commented:
Sometimes it says "auto updates" and sometimes "automated updates"
I think it would be better to standardise on just one name and having two might lead to some mistranslations

bembelimen committed:
Apply suggestions from code review

Co-authored-by: Brian Teeman <brian@teeman.net>

bembelimen commented:
Thx @brianteeman for the fixes

bembelimen committed:
Update administrator/components/com_joomlaupdate/config.xml

Co-authored-by: Brian Teeman <brian@teeman.net>

Commit:
Update notification

HLeithner committed:
Merge branch 'automated-updates/client' into feature/info-mails

bembelimen committed:
Rename auto update => automated update

Commit:
Fixed notification

bembelimen committed:
Automated update => automated updates

Commit:
Fixed notification

brianteeman commented:
When trying to save the options in order to get an update token I got the following error. I assume that is because j51.test is not a real domain and just local but still the error message should be better. (i did get an updated id however)

![image](https://github.com/user-attachments/assets/095d649a-78e7-4e0a-a5f1-7e7f9f4790f0)

brianteeman commented:
Last Checked??

I assumed that this would get a value each time I manually check for updates in the component. Do I assume incorrectly

brianteeman commented:
Post Installation Message
What do I have to do to see the new message

![image](https://github.com/user-attachments/assets/b7e8451a-7f5d-4598-b088-67994d4d9c93)

HLeithner commented:
> When trying to save the options in order to get an update token I got the following error. I assume that is because j51.test is not a real domain and just local but still the error message should be better. (i did get an updated id however)

yes you assume right, the reason is that your webserver couldn't be reached, server message is generated server side so a couple of issues cloud be have more human readable error messages.



> Last Checked??
> 
> I assumed that this would get a value each time I manually check for updates in the component. Do I assume incorrectly

No, the Update Server checks every 24 hours the health of the website, when this is done this field is updates. A indication for you that your side will get updates, this timestamp is also used in the quick icon, if it's older then 4 days you get a warning.



> Post Installation Message What do I have to do to see the new message

disable the autoupdate in the configuration, or update a joomla 5.2 side.


> Auth Token
> 
> > For access control, an auth token, that is generated in the site and is sent to the server on registration, is used.
> 
> How unique is this? What happens if I copy an entire site to a new domain.

combination of domain and token is unique, if you copy a site the same token will be used for registration, which is not a problem.

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -51,6 +63,9 @@ COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_REASON_DATABASE="Your %1$s version \"%2$s
 COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_REASON_PHP="Your PHP version \"%1$s\" is lower than \"%2$s\"."
 COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_TITLE="This site can't be updated to Joomla %1$s"
 COM_JOOMLAUPDATE_OVERVIEW="Joomla Update"
+COM_JOOMLAUPDATE_POSTINSTALL_MSG_AUTOMATED_UPDATES_ACTION="Active Automated Updates"
```
```suggestion
COM_JOOMLAUPDATE_POSTINSTALL_MSG_AUTOMATED_UPDATES_ACTION="Activate Automated Updates"
```

![image](https://github.com/user-attachments/assets/a5cff569-53e8-4fb0-bf64-9a461967efe0)

brianteeman commented:
When you disable automated updates in the options it appears to be trying to connect to the update service and deactivate it there. However the error message given is the following which doesnt make sense

![image](https://github.com/user-attachments/assets/b3a1aa18-7880-47b2-a2b6-4fd0c88f9bed)

alikon commented:
shouldn't this pr made against 6 ?

richard67 commented:
@SniperSister The update SQL scripts need to be renamed. Currently the newest update SQL in the 5.3-dev branch is "5.3.0-2025-03-14.sql", and the newest one which has already been released with a beta is "5.3.0-2025-02-22.sql".

So anything coming with the next release has to be newer than "5.3.0-2025-02-22.sql".

If we would grant updating between nightly builds to work, we would have to rename all new scripts to something newer than "5.3.0-2025-03-14.sql". But the "5.3.0-2025-03-14.sql" is not new, and it should not be modified if not absolutely necessary.

This would mean: Rename "5.3.0-2025-03-15.sql" to "5.3.0-2025-03-17.sql", rename "5.3.0-2025-01-18.sql" to "5.3.0-2025-03-15.sql", leave "5.3.0-2025-03-14.sql" as it is in the 5.3-dev branch, and create a new update SQL "5.3.0-2025-03-16.sql" which updates the guided tours from "5.3.0-2025-03-14.sql" to contain your changes.

But it seems your "5.3.0-2025-01-18.sql" shall run before the others, as it defined the new extension, and if we only grant updates between the betas, i.e. from beta 2 to beta 3, we could do it easier and modify the "5.3.0-2025-03-14.sql" like you did.

This would mean: Rename "5.3.0-2025-01-18.sql" to "5.3.0-2025-03-13.sql" and leave the other 2 as they are in your PR,

But in this case the "5.3.0-2025-03-13.sql" will not run if someone updates to beta 3 from beta 2 a 5.3 nightly build which contained already the "5.3.0-2025-03-14.sql".

Please chose what you prefer, I can help then if necessary with a PR against your branch.

SniperSister commented:
> shouldn't this pr made against 6 ?

It’s not a b/c break?

richard67 commented:
5.3 had already feature freeze, or not?

alikon commented:
but it's a new feature or it's not ? a lot of more silly pr's have been moved massively to 6
i'm  sorry still trying to understand the "criteria"

SniperSister commented:
FYI:

brianteeman commented on administrator/language/en-GB/guidedtours.joomla_whatsnew_5_3_steps.ini:
```diff
@@ -3,5 +3,7 @@
 
-COM_GUIDEDTOURS_TOUR_WHATSNEW_5_3_STEP_0_DESCRIPTION="<h3>Joomla 5.3 Full Release Notes</h3><p><a title=\"Full release notes\" href=\"https://www.joomla.org/announcements.html\" target=\"_blank\" rel=\"noopener noreferrer\">View the full release notes on joomla.org</a></p><hr><h3>Help and Information</h3><p>Many resources to help you can be found here together with additional links to more resour ...
-COM_GUIDEDTOURS_TOUR_WHATSNEW_5_3_STEP_0_TITLE="More Information and Help"
+COM_GUIDEDTOURS_TOUR_WHATSNEW_5_3_STEP_1_DESCRIPTION="<h3>Joomla 5.3 Full Release Notes</h3><p><a title=\"Full release notes\" href=\"https://www.joomla.org/announcements.html\" target=\"_blank\" rel=\"noopener noreferrer\">View the full release notes on joomla.org</a></p><hr><h3>Help and Information</h3><p>Many resources to help you can be found here together with additional links to more resour ...
+COM_GUIDEDTOURS_TOUR_WHATSNEW_5_3_STEP_1_TITLE="More Information and Help"
+COM_GUIDEDTOURS_TOUR_WHATSNEW_5_3_STEP_0_DESCRIPTION="<h1 class=\"fw-medium\">More mentionable features</h1><div class=\"row align-items-center mb-4 pt-4 border-top border-1\"><div class=\"col-md-6 col-lg-7\"><h2 class=\"mb-3\">Guided Tours Updates</h2><p>Guided Tours now has its own API. <a href=\"https://manual.joomla.org/docs/next/web-services-api/\" target=\"_blank\" rel=\"noopener noreferrer ...
+COM_GUIDEDTOURS_TOUR_WHATSNEW_5_3_STEP_0_TITLE="More mentionable features"
```
```suggestion
COM_GUIDEDTOURS_TOUR_WHATSNEW_5_3_STEP_0_DESCRIPTION="<h1 class=\"fw-medium\">More mentionable features</h1><div class=\"row align-items-center mb-4 pt-4 border-top border-1\"><div class=\"col-md-6 col-lg-7\"><h2 class=\"mb-3\">Guided Tours Updates</h2><p>Guided Tours now has its own API. <a href=\"https://manual.joomla.org/docs/next/web-services-api/\" target=\"_blank\" rel=\"noopener noreferrer\">Joomla API Documentation</a></p><p>Stay tuned for other new enhancements coming soon!</p></div></div><div class=\"row align-items-center border-top border-1 pt-4\"><div class=\"col-md-6 col-lg-5 text-center text-md-start mb-3 mb-md-0\"><img class=\"img-fluid\" src=\"media/com_guidedtours/images/5_3/filesdirectory.jpg\" alt=\"The first step of the welcome tour, showing the new media files folder\" width=\"521\" height=\"304\"></div><div class=\"col-md-6 col-lg-7 mb-3\"><h2 class=\"mb-3\">Core Updates</h2><h3 class=\"mb-3\">Organize Files in their Own Location</h3><p>Now you can see and select your stored/uploaded documents directly from the Media Manager. Find more information in the <a href=\"https://magazine.joomla.org/all-issues/december-2024/document-uploads-on-your-joomla-website\" target=\"_blank\" rel=\"noopener noreferrer\">Joomla Community Magazine</a>.</p><h3 class=\"mb-3\">View Scheduled Task History</h3><p>You now have the ability to view execution information directly in the backend without having to access log files. Look for the <strong>Execution History</strong> button located in the Scheduled Tasks page.</p><h3>Other Enhancements and Fixes</h3><ul class=\"mb-3\"><li>Template Override Plugin Comparison - adds compare functionality to plugin overrides</li><li>Tag Alias Lookup Fixed - now instead of showing the tag id number in the URL the tag name is shown</li></ul></div></div>"
COM_GUIDEDTOURS_TOUR_WHATSNEW_5_3_STEP_0_TITLE="More mentionable features"
COM_GUIDEDTOURS_TOUR_WHATSNEW_5_3_STEP_1_DESCRIPTION="<h3>Joomla 5.3 Full Release Notes</h3><p><a title=\"Full release notes\" href=\"https://www.joomla.org/announcements.html\" target=\"_blank\" rel=\"noopener noreferrer\">View the full release notes on joomla.org</a></p><hr><h3>Help and Information</h3><p>Many resources to help you can be found here together with additional links to more resources, documentation, support and how to become involved in Joomla.</p>"
COM_GUIDEDTOURS_TOUR_WHATSNEW_5_3_STEP_1_TITLE="More Information and Help"
```

richard67 replied:
The changes for the 5.3 files have been reverted, and new ones for 5.4 have been created, so this conversation is resolved.

brianteeman commented on administrator/language/en-GB/plg_actionlog_joomla.ini:
```diff
@@ -55,6 +55,7 @@ PLG_ACTIONLOG_JOOMLA_USER_RESET_COMPLETE="User <a href='{accountlink}'>{username
 PLG_ACTIONLOG_JOOMLA_USER_RESET_REQUEST="User <a href='{accountlink}'>{username}</a> requested a password reset for their account"
 PLG_ACTIONLOG_JOOMLA_USER_UNBLOCK="User <a href='{accountlink}'>{username}</a> unblocked user <a href='{itemlink}'>{title}</a>"
 PLG_ACTIONLOG_JOOMLA_USER_UPDATE="User <a href='{accountlink}'>{username}</a> updated Joomla from {oldversion} to {version}"
+PLG_ACTIONLOG_JOOMLA_SYSTEM_UPDATE="An automated process updated Joomla from {oldversion} to {version}"
```
please alpha sort

richard67 replied:
Does it really belong to the section `; Content types title` (lines 13 and below)? Or does it and maybe also the `PLG_ACTIONLOG_JOOMLA_USER` texts belong to something else?

It seems that file has different sections.

brianteeman commented on api/components/com_joomlaupdate/src/View/Updates/JsonapiView.php:
```diff
@@ -0,0 +1,135 @@
+ * @package     Joomla.API
+ * @subpackage  com_joomlaupdate
+ *
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2025 Open Source Matters, Inc. <https://www.joomla.org>
```

brianteeman commented:
> FYI: we decided to move the feature to 5.4, I’ll rebase the PR tomorrow.

all other proposed new features that did not make the 5.3 feature freeze last month were moved by the maintainers to j6

alikon commented:
i would like to clarify, i'm happy for this feature and i would like to see it in wild as soon as possible, and thanks to you all for your big effort, i'm just trying to understand, what is  the criteria, just that, we have 4.x branch , 5.2 branch , 5.3 branch, 5.4 branch , and 6 branch so once again what is the criteria ?

richard67 commented:
@brianteeman Yes, 5.4 shall not be a feature release, it shall be the bridge version to 6.0, like 3.10 was for 4 and 4.4 is for 5, and so 5.4 focuses on stable and safe updating to 6 sand of course bug fixes.

However, there are some exceptions if a new feature helps with the updating to the next major. In 3.10 we had the pre-update checker. In 4.4 we had the b/c plugin as a stub so if was available when updating to 5. And in 5.4 it makes sense to have already the infrastructure for the automated updates so these do not have to introduced when updating to 6, even if we won't use the automated update for updating to 6 because as far as I understand it will not update to new major versions, only new minors or patch versions. But having the automated updates already working on 5.4 reduces the risk when updating to 6 in my opinion.

That's why I am ok with it for going into 5.4.

richard67 commented:
> i would like to clarify, i'm happy for this feature and i would like to see it in wild as soon as possible, and thanks to you all for your big effort, i'm just trying to understand, what is the criteria, just that, we have 4.x branch , 5.2 branch , 5.3 branch, 5.4 branch , and 6 branch so once again what is the criteria ?

@alikon Se my previous comment about new features. See the road map about the rest.

richard67 commented:
Unrelated changed files shown due to the rebase will disappear after the next upmerge from 5.3-dev to 5.4-dev and then a branch update of this PR. Am working on it right now.

richard67 committed:
Merge branch '5.4-dev' into automated-updates/client

richard67 committed:
Rename update SQL scripts

richard67 committed:
Revert guided tours update SQL changes

richard67 committed:
Move image for guided tours to 5.4

richard67 committed:
Revert guided tours 5.3 language file changes

richard67 committed:
Change postinstall message condition to 5.4.0

richard67 committed:
Change postinstall message to 5.4

richard67 committed:
Add guided tours what's new in 5.4

brianteeman commented on administrator/language/en-GB/plg_quickicon_autoupdate.ini:
```diff
@@ -0,0 +1,15 @@
+; Note : All ini files need to be saved as UTF-8
+
+PLG_QUICKICON_AUTOUPDATE="Quick Icon - Joomla! Auto Update Health Notification"
+PLG_QUICKICON_AUTOUPDATE_CHECKING="Checking Auto Update &hellip;"
```
```suggestion
PLG_QUICKICON_AUTOUPDATE_CHECKING="Checking Automated Update &hellip;"
```

brianteeman commented on administrator/language/en-GB/plg_quickicon_autoupdate.ini:
```diff
@@ -0,0 +1,15 @@
+PLG_QUICKICON_AUTOUPDATE_GROUP_LABEL="Group"
+PLG_QUICKICON_AUTOUPDATE_OK="Automated Updates working."
+PLG_QUICKICON_AUTOUPDATE_OUTDATED="Automated Updates connection broken."
+PLG_QUICKICON_AUTOUPDATE_XML_DESCRIPTION="Checks the health status of auto updates and notifies you when you visit the Home Dashboard page."
```
```suggestion
PLG_QUICKICON_AUTOUPDATE_XML_DESCRIPTION="Checks the health status of automated updates and notifies you when you visit the Home Dashboard page."
```

brianteeman commented on administrator/language/en-GB/plg_quickicon_autoupdate.ini:
```diff
@@ -0,0 +1,15 @@
+; License GNU General Public License version 2 or later; see LICENSE.txt
+; Note : All ini files need to be saved as UTF-8
+
+PLG_QUICKICON_AUTOUPDATE="Quick Icon - Joomla! Auto Update Health Notification"
```
```suggestion
PLG_QUICKICON_AUTOUPDATE="Quick Icon - Joomla! Automated Update Health Notification"
```

brianteeman commented on administrator/language/en-GB/plg_quickicon_autoupdate.sys.ini:
```diff
@@ -0,0 +1,7 @@
+; License GNU General Public License version 2 or later; see LICENSE.txt
+; Note : All ini files need to be saved as UTF-8
+
+PLG_QUICKICON_AUTOUPDATE="Quick Icon - Joomla! Auto Update Health Notification"
```
```suggestion
PLG_QUICKICON_AUTOUPDATE="Quick Icon - Joomla! Automated Update Health Notification"
```

brianteeman commented on administrator/language/en-GB/plg_quickicon_autoupdate.sys.ini:
```diff
@@ -0,0 +1,7 @@
+; Note : All ini files need to be saved as UTF-8
+
+PLG_QUICKICON_AUTOUPDATE="Quick Icon - Joomla! Auto Update Health Notification"
+PLG_QUICKICON_AUTOUPDATE_XML_DESCRIPTION="Checks the health status of auto updates and notifies you when you visit the Home Dashboard page."
```
```suggestion
PLG_QUICKICON_AUTOUPDATE_XML_DESCRIPTION="Checks the health status of automated updates and notifies you when you visit the Home Dashboard page."
```

SniperSister committed:
Merge pull request #3 from richard67/5.4-dev-automated-updates-client-mod-1

[CMS PR 45143] Adapt automated updates PR to 5.4-dev branch

richard67 commented:
@SniperSister Testing instructions should be adjusted to 5.4.0-alpha1. Maybe we can find a way to test it before that, e.g. when this PR is merged and we have 5.4-dev nightlies which include this PR?

brianteeman commented on plugins/quickicon/autoupdate/autoupdate.xml:
```diff
@@ -0,0 +1,34 @@
+	<license>GNU General Public License version 2 or later; see LICENSE.txt</license>
+	<authorEmail>admin@joomla.org</authorEmail>
+	<authorUrl>www.joomla.org</authorUrl>
+	<version>5.3.0</version>
```
```suggestion
	<version>5.4.0</version>
```

brianteeman commented on plugins/quickicon/autoupdate/services/provider.php:
```diff
@@ -0,0 +1,49 @@
+     *
+     * @return  void
+     *
+     * @since   4.0.0
```
```suggestion
     * @since   __DEPLOY_VERSION__
```

brianteeman commented on plugins/webservices/joomlaupdate/joomlaupdate.xml:
```diff
@@ -0,0 +1,21 @@
+	<license>GNU General Public License version 2 or later; see LICENSE.txt</license>
+	<authorEmail>admin@joomla.org</authorEmail>
+	<authorUrl>www.joomla.org</authorUrl>
+	<version>5.3.0</version>
```
```suggestion
	<version>5.4.0</version>
```

richard67 committed:
Apply suggestions from code review

Thanks @brianteeman

Co-authored-by: Brian Teeman <brian@teeman.net>

muhme commented:
@SniperSister and everyone else, thank you in advance for including this feature in version 5.4 ❤️.
And we require a detailed test plan to confirm that automated core updates are functioning correctly on all existing sites under all circumstances

SniperSister committed:
Merge pull request #23 from joomla-projects/feature/info-mails

Draft: Feature/info mails

brianteeman commented:
> new mail templates to notify admins about successful or failed updates

Did these get missed from the pr

HLeithner commented:
I think @SniperSister branch has not been sync with the development repository yet.

obuisard commented:
I would suggest we take the Guided Tour out of this PR, because it may not be the only information we want to add to the 'What's new tour' for 5.4. I have personally kept the code/language keys/image that were created in this PR so that we can create a separate one with the tour and what may be added to it or modified.
I really appreciate the work that was done here.

brianteeman commented:
@obuisard you can always update the tour at a later date

richard67 commented:
@obuisard I would leave it as it is in this PR if this PR will not take too long to be ready for review and testing.

If we later see that this PR is still not ready and we want to add something to the welcome tour for 5.4.0, we still can remove it from this PR.

SniperSister committed:
Merge remote-tracking branch 'my-origin/automated-updates/client' into automated-updates/client

SniperSister committed:
implement code review suggestions

SniperSister committed:
Added CLI commands for registration and unregistraoitn

SniperSister committed:
Update libraries/src/Console/AutomatedUpdatesUnregisterCommand.php

Co-authored-by: Brian Teeman <brian@teeman.net>

SniperSister committed:
Update libraries/src/Console/AutomatedUpdatesRegisterCommand.php

Co-authored-by: Brian Teeman <brian@teeman.net>

SniperSister committed:
Update libraries/src/Console/AutomatedUpdatesUnregisterCommand.php

Co-authored-by: Brian Teeman <brian@teeman.net>

HLeithner committed:
Merge pull request #26 from joomla-projects/automated-updates/cli-commands

Added CLI commands for registration and unregistraoitn

SniperSister committed:
cs fix

SniperSister committed:
cs fix

brianteeman commented on administrator/components/com_admin/sql/updates/mysql/5.4.0-2025-03-17.sql:
```diff
@@ -0,0 +1,3 @@
+-- add post-installation messaged for automated updates
```
```suggestion
-- add post-installation message for automated updates
```

richard67 replied:
I've made the same suggestion to the new, combined file. Thanks.

brianteeman commented on administrator/components/com_admin/sql/updates/postgresql/5.4.0-2025-03-17.sql:
```diff
@@ -0,0 +1,3 @@
+-- add post-installation messaged for automated updates
```
```suggestion
-- add post-installation message for automated updates
```

richard67 replied:
I've made the same suggestion to the new, combined file. Thanks.

brianteeman commented on administrator/components/com_joomlaupdate/src/Model/NotificationModel.php:
```diff
@@ -0,0 +1,179 @@
+ * @package     Joomla.Administrator
+ * @subpackage  com_joomlaupdate
+ *
+ * @copyright   (C) 2012 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2025 Open Source Matters, Inc. <https://www.joomla.org>
```

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,18 +3,32 @@
 ; License GNU General Public License version 2 or later; see LICENSE.txt
 ; Note : All ini files need to be saved as UTF-8
 
+COM_JOOMLAUPDATE_AUTOUPDATE_REGISTER_SUCCESS="Registered to autoupdate service"
+COM_JOOMLAUPDATE_AUTOUPDATE_REGISTER_ERROR="Error while registering to autoupdate service: %s (%d)"
+COM_JOOMLAUPDATE_AUTOUPDATE_UNREGISTER_SUCCESS="Unregistered from autoupdate service"
+COM_JOOMLAUPDATE_AUTOUPDATE_UNREGISTER_ERROR="Error while unregistering from autoupdate service: %s (%d)"
```
```suggestion
COM_JOOMLAUPDATE_AUTOUPDATE_REGISTER_ERROR="Error while registering to autoupdate service: %s (%d)"
COM_JOOMLAUPDATE_AUTOUPDATE_REGISTER_SUCCESS="Registered to autoupdate service"
COM_JOOMLAUPDATE_AUTOUPDATE_UNREGISTER_ERROR="Error while unregistering from autoupdate service: %s (%d)"
COM_JOOMLAUPDATE_AUTOUPDATE_UNREGISTER_SUCCESS="Unregistered from autoupdate service"
```

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,18 +3,32 @@
 COM_JOOMLAUPDATE_CONFIG_SOURCES_DESC="Configure where Joomla gets its update information from."
 COM_JOOMLAUPDATE_CONFIG_SOURCES_LABEL="Update Source"
 COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_CUSTOM="Custom URL"
+COM_JOOMLAUPDATE_CONFIG_UPDATE_LAST_CHECK_LABEL="Last Checked"
+COM_JOOMLAUPDATE_CONFIG_UPDATE_LAST_CHECK_DESC="The last time the automated updates server could reach this site. If this date is too old, something is wrong with the automated updates and you should re-initiate the process by re-saving the configuration."
```
```suggestion
COM_JOOMLAUPDATE_CONFIG_UPDATE_LAST_CHECK_DESC="The last time the automated updates server could reach this site. If this date is too old, something is wrong with the automated updates and you should re-initiate the process by re-saving the configuration."
COM_JOOMLAUPDATE_CONFIG_UPDATE_LAST_CHECK_LABEL="Last Checked"

```

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -86,6 +103,10 @@ COM_JOOMLAUPDATE_UPDATE_LOG_UNINSTALL_EXTENSIONS="Uninstalling extensions"
 COM_JOOMLAUPDATE_UPDATE_LOG_UPLOAD="Uploading update file"
 COM_JOOMLAUPDATE_UPDATE_LOG_URL="Downloading update file from %s."
 COM_JOOMLAUPDATE_UPDATE_LOGGING_TEST_FAIL="Logging does not work. Make sure the log folder is writable, and try again. The logging test failed with message: %s"
+COM_JOOMLAUPDATE_UPDATE_SUCCESS_MAIL_BODY="This email IS NOT sent by the Joomla! project. It is sent automatically by your own site,\n{SITENAME} - {URL} \n================================================================================\nUPDATE INFORMATION\n================================================================================\nYour site has been successfully automatically updated from { ...
+COM_JOOMLAUPDATE_UPDATE_SUCCESS_MAIL_SUBJECT="Joomla! update failed for {SITENAME} – {URL}"
+COM_JOOMLAUPDATE_UPDATE_FAILED_MAIL_BODY="This email IS NOT sent by the Joomla! project. It is sent automatically by your own site,\n{SITENAME} - {URL} \n================================================================================\nUPDATE INFORMATION\n================================================================================\nYour site could not be updated from {OLDVERSION} to {NEWVERSI ...
+COM_JOOMLAUPDATE_UPDATE_FAILED_MAIL_SUBJECT="Joomla! successfully updated at {SITENAME} – {URL}"
```
```suggestion
COM_JOOMLAUPDATE_UPDATE_FAILED_MAIL_BODY="This email IS NOT sent by the Joomla! project. It is sent automatically by your own site,\n{SITENAME} - {URL} \n================================================================================\nUPDATE INFORMATION\n================================================================================\nYour site could not be updated from {OLDVERSION} to {NEWVERSION}.\nPlease check the logfile of com_joomlaupdate for further debugging information.\nThis email is sent to you by your site.\nThe Joomla! project will never contact you directly.\n================================================================================\nWHY AM I RECEIVING THIS EMAIL?\n================================================================================\nThis email has been automatically sent by Joomla!, the software which powers your site.\nYou have enabled automatic updates in your site and trying to install such an update triggers this message.\nIf you do not understand what Joomla! is and what you need to do please do not contact the Joomla! project.\nThey are NOT sending you this email and they cannot help you. Instead, please contact the person who built or manages your site.\n================================================================================\nWHO SENT ME THIS EMAIL?\n================================================================================\nThis email is sent to you by your own site, {SITENAME}"
COM_JOOMLAUPDATE_UPDATE_FAILED_MAIL_SUBJECT="Joomla! updated at {SITENAME} – {URL}"
COM_JOOMLAUPDATE_UPDATE_SUCCESS_MAIL_BODY="This email IS NOT sent by the Joomla! project. It is sent automatically by your own site,\n{SITENAME} - {URL} \n================================================================================\nUPDATE INFORMATION\n================================================================================\nYour site has been successfully automatically updated from {OLDVERSION} to {NEWVERSION}.\nPlease check your site for any unexpected behavior or malfunctions.\nThis email is sent to you by your site.\nThe Joomla! project will never contact you directly.\n================================================================================\nWHY AM I RECEIVING THIS EMAIL?\n================================================================================\nThis email has been automatically sent by Joomla!, the software which powers your site.\nYou have enabled automatic updates in your site and the installation of such an update triggers this message.\nIf you do not understand what Joomla! is and what you need to do please do not contact the Joomla! project.\nThey are NOT sending you this email and they cannot help you. Instead, please contact the person who built or manages your site.\n================================================================================\nWHO SENT ME THIS EMAIL?\n================================================================================\nThis email is sent to you by your own site, {SITENAME}"
COM_JOOMLAUPDATE_UPDATE_SUCCESS_MAIL_SUBJECT="Joomla! update failed for {SITENAME} – {URL}"
```

brianteeman replied:
Alpha sorted and removed the redundant word successfully

brianteeman commented on api/components/com_joomlaupdate/src/View/Healthcheck/JsonapiView.php:
```diff
@@ -0,0 +1,78 @@
+ * @package     Joomla.API
+ * @subpackage  com_joomlaupdate
+ *
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2025 Open Source Matters, Inc. <https://www.joomla.org>
```

brianteeman commented on api/components/com_joomlaupdate/src/View/Notification/JsonapiView.php:
```diff
@@ -0,0 +1,59 @@
+ * @package     Joomla.API
+ * @subpackage  com_joomlaupdate
+ *
+ * @copyright   (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
```
```suggestion
 * @copyright   (C) 2025 Open Source Matters, Inc. <https://www.joomla.org>
```

richard67 commented:
@SniperSister Please rename the update SQL scripts to something newer than the "5.4.0-2025-04-23.sql" currently present in the 5.4-dev branch, e.g. rename "5.4.0-2025-03-16.sql" to "5.4.0-2025-05-09.sql", "5.4.0-2025-03-17.sql" to "5.4.0-2025-05-10.sql" and "5.4.0-2025-03-18.sql" to "5.4.0-2025-05-11.sql".

richard67 commented:
@SniperSister P.S.: ... and add the check of the postinstall message to the testing instructions. Thanks in advance.

HLeithner committed:
Update administrator/language/en-GB/com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

HLeithner committed:
Update api/components/com_joomlaupdate/src/View/Healthcheck/JsonapiView.php

Co-authored-by: Brian Teeman <brian@teeman.net>

HLeithner committed:
Update api/components/com_joomlaupdate/src/View/Notification/JsonapiView.php

Co-authored-by: Brian Teeman <brian@teeman.net>

HLeithner committed:
Combine sql update files

HLeithner commented:
> @SniperSister Please rename the update SQL scripts to something newer than the "5.4.0-2025-04-23.sql" currently present in the 5.4-dev branch, e.g. rename "5.4.0-2025-03-16.sql" to "5.4.0-2025-05-09.sql", "5.4.0-2025-03-17.sql" to "5.4.0-2025-05-10.sql" and "5.4.0-2025-03-18.sql" to "5.4.0-2025-05-11.sql".

I combined the 3 files to one file with todays date

richard67 commented on installation/sql/mysql/base.sql:
```diff
@@ -391,6 +392,7 @@ INSERT INTO `#__extensions` (`package_id`, `name`, `type`, `element`, `folder`,
 (0, 'plg_webservices_contact', 'plugin', 'contact', 'webservices', 0, 1, 1, 0, 1, '', '{}', '', 3, 0),
 (0, 'plg_webservices_content', 'plugin', 'content', 'webservices', 0, 1, 1, 0, 1, '', '{}', '', 4, 0),
 (0, 'plg_webservices_installer', 'plugin', 'installer', 'webservices', 0, 1, 1, 0, 1, '', '{}', '', 5, 0),
+(0, 'plg_webservices_joomlaupdate', 'plugin', 'joomlaupdate', 'webservices', 0, 1, 1, 0, 1, '', '{}', '', 6, 0),
```
The ordering of the other webservices plugins should be adapted to that. Now we would have a duplicate 6. But as there are already other records with duplicate ordering (7), we can clean that up with a follow up PR.

richard67 replied:
See https://github.com/SniperSister/joomla-cms/pull/4 for fixing this.

richard67 commented on installation/sql/postgresql/base.sql:
```diff
@@ -397,6 +398,7 @@ INSERT INTO "#__extensions" ("package_id", "name", "type", "element", "folder",
 (0, 'plg_webservices_contact', 'plugin', 'contact', 'webservices', 0, 1, 1, 0, 1, '', '{}', '', 3, 0),
 (0, 'plg_webservices_content', 'plugin', 'content', 'webservices', 0, 1, 1, 0, 1, '', '{}', '', 4, 0),
 (0, 'plg_webservices_installer', 'plugin', 'installer', 'webservices', 0, 1, 1, 0, 1, '', '{}', '', 5, 0),
+(0, 'plg_webservices_joomlaupdate', 'plugin', 'joomlaupdate', 'webservices', 0, 1, 1, 0, 1, '', '{}', '', 5, 0),
```
```suggestion
(0, 'plg_webservices_joomlaupdate', 'plugin', 'joomlaupdate', 'webservices', 0, 1, 1, 0, 1, '', '{}', '', 6, 0),
```
Should be the same as in the MySQL file.

The same comment also applies here, the ordering needs to be cleaned up so it is unique for all the webservices plugins.

But that can be done with a follow up PR.

richard67 replied:
See https://github.com/SniperSister/joomla-cms/pull/4 for fixing this.

richard67 committed:
Fix ordering of webservices plugins

richard67 commented on administrator/components/com_admin/sql/updates/mysql/5.4.0-2025-05-10.sql:
```diff
@@ -0,0 +1,26 @@
+('com_joomlaupdate.update.success', 'com_joomlaupdate', '', 'COM_JOOMLAUPDATE_UPDATE_SUCCESS_MAIL_SUBJECT', 'COM_JOOMLAUPDATE_UPDATE_SUCCESS_MAIL_BODY', '', '', '{"tags":["newversion","oldversion","sitename","url"]}'),
+('com_joomlaupdate.update.failed', 'com_joomlaupdate', '', 'COM_JOOMLAUPDATE_UPDATE_FAILED_MAIL_SUBJECT', 'COM_JOOMLAUPDATE_UPDATE_FAILED_MAIL_BODY', '', '', '{"tags":["newversion","oldversion","sitename","url"]}');
+
+-- add post-installation messaged for automated updates
```
```suggestion
-- add post-installation message for automated updates
```

richard67 commented on administrator/components/com_admin/sql/updates/postgresql/5.4.0-2025-05-10.sql:
```diff
@@ -0,0 +1,26 @@
+('com_joomlaupdate.update.success', 'com_joomlaupdate', '', 'COM_JOOMLAUPDATE_UPDATE_SUCCESS_MAIL_SUBJECT', 'COM_JOOMLAUPDATE_UPDATE_SUCCESS_MAIL_BODY', '', '', '{"tags":["newversion","oldversion","sitename","url"]}'),
+('com_joomlaupdate.update.failed', 'com_joomlaupdate', '', 'COM_JOOMLAUPDATE_UPDATE_FAILED_MAIL_SUBJECT', 'COM_JOOMLAUPDATE_UPDATE_FAILED_MAIL_BODY', '', '', '{"tags":["newversion","oldversion","sitename","url"]}');
+
+-- add post-installation messaged for automated updates
```
```suggestion
-- add post-installation message for automated updates
```

SniperSister committed:
Merge pull request #4 from richard67/5.4-dev-automated-updates-client-mod-2

[CMS PR 45143] Fix ordering of webservices plugins

SniperSister committed:
adjust string sorting

SniperSister committed:
Apply suggestions from code review

Co-authored-by: Richard Fath <richard67@users.noreply.github.com>
Co-authored-by: Brian Teeman <brian@teeman.net>

SniperSister committed:
Merge branch 'automated-updates/client' of https://github.com/SniperSister/joomla-cms into automated-updates/client

richard67 committed:
Merge branch '5.4-dev' into automated-updates/client

SniperSister committed:
fix missing import

SniperSister commented:
I've merged the suggestions by @richard67 and @brianteeman  (thank you guys) and fixed a missing `use` statement that was discovered during internal testing

richard67 commented:
> and fixed a missing `use` statement that was discovered during internal testing

@SniperSister I don't see a fix like that in the latest commits. Could youz double check if that statement is there?

SniperSister committed:
Merge branch 'automated-updates/client' of https://github.com/SniperSister/joomla-cms into automated-updates/client

SniperSister commented:
@richard67 see https://github.com/joomla/joomla-cms/pull/45143/commits/7646135698de030f15aad6d32fe94ce2b7743dea

richard67 committed:
Merge branch '5.4-dev' into automated-updates/client

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,18 +3,33 @@
 ; License GNU General Public License version 2 or later; see LICENSE.txt
 ; Note : All ini files need to be saved as UTF-8
 
+COM_JOOMLAUPDATE_AUTOUPDATE_REGISTER_ERROR="Error while registering to autoupdate service: %s (%d)"
```
error and success messages should be complete sentence ie they should end with a period

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -3,18 +3,33 @@
 COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_DEFAULT="Default"
 COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_LABEL="Update Channel"
 COM_JOOMLAUPDATE_CONFIG_UPDATESOURCE_NEXT="Joomla Next"
+COM_JOOMLAUPDATE_CONFIG_UPDATE_TOKEN_DESC="This is the update token that this site uses to authenticate with the Joomla Update Server. It is used to ensure that only your site can update itself."
```
please can we be consistent with the terminology

Joomla Update Server.
automated updates server

brianteeman commented on administrator/language/en-GB/com_joomlaupdate.ini:
```diff
@@ -51,6 +66,9 @@ COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_REASON_DATABASE="Your %1$s version \"%2$s
 COM_JOOMLAUPDATE_NODOWNLOAD_EMPTYSTATE_TITLE="This site can't be updated to Joomla %1$s"
 COM_JOOMLAUPDATE_OVERVIEW="Joomla Update"
+COM_JOOMLAUPDATE_POSTINSTALL_MSG_AUTOMATED_UPDATES_ACTION="Activate Automated Updates"
+COM_JOOMLAUPDATE_POSTINSTALL_MSG_AUTOMATED_UPDATES_DESCRIPTION="Joomla 5.4 now supports Automated updates for better security and performance. Select \"Activate Automated Updates\" to stay up to date effortlessly!"
```
how does this feature improve performance?

HLeithner replied:
Feel free to make a better suggestion. I think the text just should motivate to activate the feature, security is the main reason but since we do all minor updates it's not only security.

brianteeman replied:
to say it improves performance when it doesn't is a lie. that is why I asked the question if it does improve performance

HLeithner replied:
than also "better security" is a lie, we don't know what will be better in the next joomla version. happy with any suggestion

brianteeman replied:
1. Stay secure and up to date effortlessly with Automated Updates that save time, reduce risk, and keep your system running smoothly.

2. Automated Updates helps minimize security vulnerabilities, and reduce manual maintenance by seamlessly delivering the latest features and fixes.

3. With Automated Updates you can keep your Joomla site secure and up-to-date—automatically, reliably, effortlessly.

4. Automated Updates streamline site maintenance by ensuring Joomla stays current, reducing security risks and compatibility issues without manual intervention."

SniperSister committed:
update i18n strings

SniperSister committed:
Merge branch 'automated-updates/client' of https://github.com/SniperSister/joomla-cms into automated-updates/client

SniperSister committed:
fix notification subjects

brianteeman committed:
Quickicon Link

This small PR ensures that the link on the quickicon goes directly to the automated updates tab of the joomla update component options

SniperSister committed:
Merge pull request #27 from brianteeman/patch-1

Quickicon Link

brianteeman committed:
Update plg_quickicon_autoupdate.ini

I understand why the string "working" was chosen but I think it is better to simply say "enabled"

we know it is working as there are multiple different messages to show there is a problem (Unknown, disabled and outdated)

IT would be more consistent with the rest of Joomla to say enabled AND someone might think that working = joomla update in progress

SniperSister committed:
Merge pull request #28 from brianteeman/patch-2

Update plg_quickicon_autoupdate.ini

SniperSister committed:
Added mail template i18n strings

SniperSister committed:
Update administrator/language/en-GB/com_joomlaupdate.ini

Co-authored-by: Brian Teeman <brian@teeman.net>

SniperSister committed:
Apply suggestions from code review

Co-authored-by: Brian Teeman <brian@teeman.net>

SniperSister committed:
Merge pull request #30 from joomla-projects/automated-updates/mailtemplate-strings

Added mail template i18n strings

richard67 committed:
Merge branch '5.4-dev' into automated-updates/client

zero-24 commented on administrator/components/com_joomlaupdate/postinstall/autoupdate.php:
```diff
@@ -0,0 +1,62 @@
+function com_joomlaupdate_postinstall_autoupdate_action(): void
+{
+    $db = Factory::getContainer()->get(DatabaseInterface::class);
+    // Get extension row
```
```suggestion

    // Get extension row
```

zero-24 commented on administrator/components/com_joomlaupdate/src/Model/NotificationModel.php:
```diff
@@ -0,0 +1,179 @@
+ */
+final class NotificationModel extends BaseDatabaseModel
+{
+    public function sendNotification($type, $oldVersion): void
```
Are we missing intentional the doc block here?

zero-24 replied:
```suggestion
    /**
     * Sends the update notification to the specificly configured emails and superusers 
     *
     * @param  string  $type            The type of notification to send. This is the last key for the mail template
     * @param  string  $oldVersion  The old version from before the update
     *
     * @return  void
     *
     * @since   __DEPLOY_VERSION__
     */
    public function sendNotification($type, $oldVersion): void
```

zero-24 commented on api/components/com_joomlaupdate/src/View/Notification/JsonapiView.php:
```diff
@@ -0,0 +1,59 @@
+class JsonapiView extends BaseApiView
+{
+    /**
+     * Generates the health check output
```
```suggestion
     * Generates the health check output and sends notification about the finalization
```

zero-24 commented on api/components/com_joomlaupdate/src/View/Updates/JsonapiView.php:
```diff
@@ -0,0 +1,136 @@
+        return $this->getDocument()->render();
+    }
+
+    public function finalizeUpdate($fromVersion)
```
```suggestion
    /**
     * Run the finalize update steps 
     *
     * @param string $fromVersion The from version
     *
     * @return string  The rendered data
     *
     * @since  __DEPLOY_VERSION__
     */
    public function finalizeUpdate($fromVersion)
```

zero-24 commented on plugins/extension/joomlaupdate/src/Extension/Joomlaupdate.php:
```diff
@@ -46,29 +51,72 @@ final class Joomlaupdate extends CMSPlugin implements SubscriberInterface
+
+        $data = new Registry($table->params);
+
+        $autoupdateStatus         = (int)$data->get('autoupdate');
```
```suggestion
        $autoupdateStatus         = (int) $data->get('autoupdate');
```

zero-24 commented on plugins/extension/joomlaupdate/src/Extension/Joomlaupdate.php:
```diff
@@ -46,29 +51,72 @@ final class Joomlaupdate extends CMSPlugin implements SubscriberInterface
+        $data = new Registry($table->params);
+
+        $autoupdateStatus         = (int)$data->get('autoupdate');
+        $autoupdateRegisterStatus = (int)$data->get('autoupdate_status');
```
```suggestion
        $autoupdateRegisterStatus = (int) $data->get('autoupdate_status');
```

zero-24 commented on plugins/quickicon/autoupdate/services/provider.php:
```diff
@@ -0,0 +1,49 @@
+use Joomla\Event\DispatcherInterface;
+use Joomla\Plugin\Quickicon\Autoupdate\Extension\Autoupdate;
+
+return new class () implements ServiceProviderInterface {
```
```suggestion
return new class () implements ServiceProviderInterface
{
```

richard67 replied:
This suggestion was wrong and broke PHPCS. So I've allowed myself to change it back (together with 2 other PHPCS fixes).

zero-24 replied:
thanks

zero-24 commented on plugins/webservices/joomlaupdate/src/Extension/Joomlaupdate.php:
```diff
@@ -0,0 +1,69 @@
+    {
+        $router = $event->getRouter();
+
+        $defaults    = ['component' => 'com_joomlaupdate', 'public' => true];
```
```suggestion
        $defaults = ['component' => 'com_joomlaupdate', 'public' => true];
```

zero-24 commented on plugins/webservices/joomlaupdate/src/Extension/Joomlaupdate.php:
```diff
@@ -0,0 +1,69 @@
+    public function onBeforeApiRoute(BeforeApiRouteEvent $event): void
+    {
+        $router = $event->getRouter();
+
```
```suggestion
```

zero-24 commented on plugins/webservices/joomlaupdate/src/Extension/Joomlaupdate.php:
```diff
@@ -0,0 +1,69 @@
+     */
+    public function onBeforeApiRoute(BeforeApiRouteEvent $event): void
+    {
+        $router = $event->getRouter();
```
```suggestion
        $router  = $event->getRouter();
```

zero-24 reviewed (commented):
Thanks for this great feature, do not have the time to do intensive testing now but please accept my code review suggestings about the smal things i found while taking a look into the code. Thanks 👍

SniperSister committed:
Apply suggestions from code review

Co-authored-by: Tobias Zulauf <zero-24@users.noreply.github.com>

SniperSister committed:
Fix docblock

SniperSister committed:
Merge branch 'automated-updates/client' of https://github.com/SniperSister/joomla-cms into automated-updates/client

SniperSister committed:
fix site url

richard67 commented on api/components/com_joomlaupdate/src/View/Updates/JsonapiView.php:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    /**
+     * Run the finalize update steps 
```
```suggestion
     * Run the finalize update steps
```
PHPCS 1 of 3

richard67 commented on plugins/webservices/joomlaupdate/src/Extension/Joomlaupdate.php:
```diff
@@ -0,0 +1,68 @@
+    public function onBeforeApiRoute(BeforeApiRouteEvent $event): void
+    {
+        $router  = $event->getRouter();
+        $defaults = ['component' => 'com_joomlaupdate', 'public' => true];
```
```suggestion
        $router   = $event->getRouter();
        $defaults = ['component' => 'com_joomlaupdate', 'public' => true];
```
PHPCS 2 of 3

richard67 commented on plugins/quickicon/autoupdate/services/provider.php:
```diff
@@ -0,0 +1,50 @@
+use Joomla\Event\DispatcherInterface;
+use Joomla\Plugin\Quickicon\Autoupdate\Extension\Autoupdate;
+
+return new class () implements ServiceProviderInterface
+{
```
```suggestion
return new class () implements ServiceProviderInterface {
```
PHPCS 3 of 3

richard67 committed:
Fix PHPCS

brianteeman committed:
automated_updates_email

All the fields except for the admin email are hidden when the update source is not default and level is stable. 

I am assuming it is an oversight that the admin email field was not hidden

brianteeman committed:
Update com_joomlaupdate.ini

space should not be used in a language string for padding. And its not needed anyway as the icon already has me-2 class which is sufficient

HLeithner committed:
Merge pull request #35 from brianteeman/patch-2

Update com_joomlaupdate.ini

brianteeman committed:
Update com_joomlaupdate.ini

changed autoupdate to automated update for consistency

SniperSister committed:
Merge pull request #36 from brianteeman/patch-2

Update com_joomlaupdate.ini

SniperSister committed:
Merge pull request #33 from brianteeman/patch-1

automated_updates_email

SniperSister committed:
Merge branch 'automated-updates/client' of https://github.com/SniperSister/joomla-cms into automated-updates/client

SniperSister committed:
Check stability and channel in quickicon status output

SniperSister committed:
respect user time in last_check-Date

brianteeman committed:
Update NotificationModel.php

Spelling and spaces

SniperSister committed:
Merge pull request #37 from brianteeman/patch-1

Update NotificationModel.php

brianteeman committed:
Update guidedtours.joomla_whatsnew_5_4.ini

Remove redundant title text on the button - it already has perfectly accessible discernible text

Remove target = _blank - this obviously was an oversight as it sucks for accessibility and should not be used

SniperSister committed:
Merge pull request #38 from brianteeman/patch-1

Update guidedtours.joomla_whatsnew_5_4.ini

SniperSister committed:
cs fix

SniperSister committed:
cs fix

MacJoom commented:
Partly successfully tested
- Updated an existing Website from 5.3.0 to 5.4.103
- Make sure you are on Update Channel default and Minimum Stability: Stable,
- Change in Joomla Update to Update Channel default and Minimum Stability Stable,  enabled Autoupdate
- Upon Save & Close first time: 
![image](https://github.com/user-attachments/assets/f5b33054-cd5d-43a1-b37d-0452e57c5b8f)
- Save & Close second time (no changes made), OK (Registered to automated update service.)
-  Automated Update was done almost immedatly, Email Sent

HLeithner commented:
> * oomla Update to Update Channel default and Minimum Stability Stable,  enabled Autoupdate

was caching activated for this site?

MacJoom commented:
> > * oomla Update to Update Channel default and Minimum Stability Stable,  enabled Autoupdate
> 
> was caching activated for this site?

Yes - without caching (conservative & progressive) Save & Close works every time.

HLeithner commented:
> > > * oomla Update to Update Channel default and Minimum Stability Stable,  enabled Autoupdate
> > 
> > 
> > was caching activated for this site?
> 
> Yes - without caching (conservative & progressive) Save & Close works every time.

ok make sense because joomla caches the component options and we likely load it from the cache in the next second, we will see how we can solve this. Cache invalidation would be the best case but might be tricky. As alternative we could load the information directly...

HLeithner commented:
I added the 2 known issues to the pr description and would like to postpone the fix for this to a follow up pr to get this merged into the alpha1 so it can be tested easier.

MacJoom commented:
I have tested this item :white_check_mark: successfully on b25bba3bf60c27b62dfbedf2ebaba01150d31160<br /><br />Works! From existing 5.3.0 to 5.4.103 and automatically to 5.4.104 <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/45143">issues.joomla.org/tracker/joomla-cms/45143</a>.</sub>

SniperSister committed:
check update requirements in healthcheck and get update endpoints

SniperSister committed:
add cache purger

SniperSister committed:
cs fix

SniperSister committed:
fix email

SniperSister commented:
> I added the 2 known issues to the pr description and would like to postpone the fix for this to a follow up pr to get this merged into the alpha1 so it can be tested easier.

The known issues have been resolved

SniperSister committed:
fix default config state

richard67 commented:
I have tested this item :white_check_mark: successfully on b25bba3bf60c27b62dfbedf2ebaba01150d31160<br /><br />I've successfully tested that it works, also when the site is set to offline.

In addition, I've tested updating a 5.3 to the patched package of this PR with MySQL and PostgreSQL to see if the update SQL scripts are right.

Finally I've downloaded the 5.4.105 packages used for the test and have verified that they don't contain any unexpected differences to the patched packages for this PR, except of the wanted patch for the update site and the version and that it is behind the 5.4-dev branch with some changes, but that is not related to the functionality added by this PR.

What is not nice is that if you use the button in the Guided Tour for activating the auto updates, the settings show up for a short moment, but then you are back in the guided tours, and there is no visual indication if it has worked or not until you leave the guided tours and check the quick icon or the settings.

But I think this can be fixed with a follow up PR.

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/45143">issues.joomla.org/tracker/joomla-cms/45143</a>.</sub>

muhme committed:
Merge branch '5.4-dev' into automated-updates/client

richard67 committed:
Merge branch '5.4-dev' into automated-updates/client

muhme commented:
A big thank you to all contributers and testers!

richard67 commented:
From me a big thank you, too.
